x-pack/legacy/plugins/watcher/server/np_ready/lib/call_with_request_factory.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ElasticsearchServiceSetup } from 'src/core/server';
+import { once } from 'lodash';
+import { elasticsearchJsPlugin } from './elasticsearch_js_plugin';
+
+const callWithRequest = once((elasticsearchService: ElasticsearchServiceSetup) => {
+  const config = { plugins: [elasticsearchJsPlugin] };
+  return elasticsearchService.createClient('watcher', config);
+});
+
+export const callWithRequestFactory = (
```

Comment:
@rudolf Would you cast an eye on my use of `callAsCurrent` user here. Creating a client instance per request is the correct way to use this API? Considering that the param it takes in is called request 😅 ?